### PR TITLE
End to end test

### LINKED ISSUE #1711: End-to-end test for email sending
we need to prevent failures like #1709 somehow

### PR DESCRIPTION
**Pull request by opqdonut**
closes #1711

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## Testing
- [x] valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: src/clj/rems/api/workflows.clj [hunk 1]
```diff
@@ -10,7 +10,7 @@
 (s/defschema CreateWorkflowCommand
   {:organization s/Str
    :title s/Str
-   :type s/Keyword
+   :type s/Keyword ;; TODO document!
```
Comment (opqdonut) [inline src/clj/rems/api/workflows.clj]: I'll follow up with this

File: test/clj/rems/api/test_end_to_end.clj [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+              (is (.contains (:subject mail) "approved"))
+              (is (.startsWith (:body mail) "Dear E2E Applicant,")))
+            (reset! email-atom []))
+
+          (testing "entitlement granted"
+            (let [[entitlement & others] (api-call :get (str "/api/entitlements?user=" applicant-id) nil
+                                                   api-key owner-id)]
+              (is (empty? others))
+              (is (= resource-ext-id (:resource entitlement)))
+              (is (not (:end entitlement)))))
```
Comment (opqdonut) [inline test/clj/rems/api/test_end_to_end.clj]: should we test the entitlement POST here too?

Comment (luontola) [reply test/clj/rems/api/test_end_to_end.clj]: You mean when the entitlements are posted to another system? Sounds like a good thing to test all integrations.

Comment (opqdonut) [reply test/clj/rems/api/test_end_to_end.clj]: I'll add it

Comment (luontola) [reply test/clj/rems/api/test_end_to_end.clj]: A fake email server would also be more end-to-end than mocking the email API.

Comment (opqdonut) [reply test/clj/rems/api/test_end_to_end.clj]: yeah, you're right, I'll add a TODO about that

**Pull request merged**